src/main.py

```diff
@@ -47,25 +44,12 @@ def main():
     total_patients = 100
     treated_patients = 20
     medical_data = generate_dataset(total_patients, treated_patients)
-    logger.debug(medical_data[medical_data['Treatment'] == 1])
-    matched_pairs = methods.nnm2(medical_data, replacement=1, caliper=0.02, k=1)
-
-    #logger.debug(f'length of matched pairs: {matched_pairs}')
+    matched_pairs = methods.nnm2(medical_data, replacement=True, caliper=0.02, k=1)
     
-    count = 1
-  
-    for row in matched_pairs:
-        logging.debug(f"{count}\n")
-        count += 1
-        logging.debug(f"Treated Patient:\n{row[0]},\nMatched Patient(s):\n{row[1]}\n")
-    output_file = "data\\output.txt"  # Output file
-
-    with open(output_file, "w") as f:
-        count = 0
+    output_file = "data\\output.csv"  # Output file
```

Comment:
`data` folder is mostly for the input data files and sometimes for the intermediate files. All output file should be kept in the `build` folder. 
Try to give a meaningful names for all you files. For example this one can be  `matched_pairs.csv`
If folder, when you are trying to save a file does not exist in the repository (for example `src` always exists but `build` does not), than you need to check it and create the folder, before writing the file. Otherwise, anyone who clone the repo and run the project will get an error `FileNotFoundError: [Errno 2] No such file or directory: 'data\\output.csv'`